### PR TITLE
docs(logging): add lib/log conventions

### PR DESCRIPTION
**Pull request by burmudar**
Some conventions I have been thinking about while doing some of the logging updates.

## Test plan
Markdown looks good

### CODE REVIEW

File: doc/dev/how-to/add_logging.md [hunk 1]
```diff
@@ -180,3 +180,23 @@ func TestFooBar(t *testing.T) {
   logs := exportLogs()
 }
 ```
+## Conventions
+* The name of the scope should follow CamelCase
+* The logger parameter should either be after `ctx context.Context` or be the first parameter
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:185]:
```suggestion
* The logger parameter should either be immediately after `ctx context.Context`, or be the first parameter
```

File: doc/dev/how-to/add_logging.md [hunk 2]
```diff
@@ -180,3 +180,23 @@ func TestFooBar(t *testing.T) {
   logs := exportLogs()
 }
 ```
+## Conventions
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:183]:
```suggestion

## Conventions

```

File: doc/dev/how-to/add_logging.md [hunk 3]
```diff
@@ -180,3 +180,23 @@ func TestFooBar(t *testing.T) {
...
+
+    subLogger.Info("starting up")
+
+    if err := s.Start(); err != nil {
+        subLogger.Error(err)
+    }
+
+    subLogger.Info("done")
+}
+```
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:201]:
```suggestion

```go
func (s *Service) MyService(ctx context.Context, logger log.Logger) {
    subLogger := logger.With(log.Int("id", s.ID), log.String("service", s.Name"))

    subLogger.Info("starting up")

    if err := s.Start(); err != nil {
        subLogger.Error(err)
    }

    subLogger.Info("done")
}
```

```

Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:201]: I can't seem to leave a suggestion here lol but I think you might want to add spaces between the codeblock, docsite (`sg run docsite`) struggles with lack of spacing sometimes

File: doc/dev/how-to/add_logging.md [hunk 4]
```diff
@@ -180,3 +180,23 @@ func TestFooBar(t *testing.T) {
   logs := exportLogs()
 }
 ```
+## Conventions
+* The name of the scope should follow CamelCase
+* The logger parameter should either be after `ctx context.Context` or be the first parameter
+* Messages of log lines should be in lowercase ex. `log.Info("this is my lowercase log line")`
+* Capatalize each word when the log line message is not a sentence and referring to a func/component ex. `logger.Error("component.update"` prefer to capatalize each word
+* In a particular scope some fields might be repeatedly emitted. Consider creating a sub Logger in the scope by using `logger.With(...fields)`.
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:188]: I think we can move this point to https://docs.sourcegraph.com/dev/how-to/add_logging#handling-loggers , there's a similar point in https://docs.sourcegraph.com/dev/how-to/add_logging#attaching-context (at the very end) - maybe we could have a section called "Sub-loggers"?

File: doc/dev/how-to/add_logging.md [hunk 5]
```diff
@@ -180,3 +180,23 @@ func TestFooBar(t *testing.T) {
   logs := exportLogs()
 }
 ```
+## Conventions
+* The name of the scope should follow CamelCase
+* The logger parameter should either be after `ctx context.Context` or be the first parameter
+* Messages of log lines should be in lowercase ex. `log.Info("this is my lowercase log line")`
+* Capatalize each word when the log line message is not a sentence and referring to a func/component ex. `logger.Error("component.update"` prefer to capatalize each word
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:187]: I think we can add a new H2 section, "Writing log messages", before "development usage", dedicated to advice like this!

File: doc/dev/how-to/add_logging.md (hunk 1 above)
Comment (burmudar) [reply doc/dev/how-to/add_logging.md:185]: Fixed!

File: doc/dev/how-to/add_logging.md (hunk 2 above)
Comment (burmudar) [reply doc/dev/how-to/add_logging.md:183]: Fixed

File: doc/dev/how-to/add_logging.md (hunk 5 above)
Comment (burmudar) [reply doc/dev/how-to/add_logging.md:187]: good point! Moved 👍🏼

File: doc/dev/how-to/add_logging.md (hunk 4 above)
Comment (burmudar) [reply doc/dev/how-to/add_logging.md:188]: Added it!

File: doc/dev/how-to/add_logging.md [hunk 6]
```diff
@@ -133,6 +192,7 @@ This format omits fields like OpenTelemetry Resource and renders certain field t
 
 Additionally, in `SRC_DEVELOPMENT=true` using `log.Scoped` without calling `log.Init` will panic (in production, a no-op logger will be returned).
 
+
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:195]:
```suggestion
```

File: doc/dev/how-to/add_logging.md [hunk 7]
```diff
@@ -180,3 +240,11 @@ func TestFooBar(t *testing.T) {
   logs := exportLogs()
 }
 ```
+
+When writing a test, ensure that `logtest.Scope` in the tightest scope possible. This ensures that if a test fails, that the logging is closely tied to the test that failed. Especially if you testcase has sub tests with `t.Run`, prefer to created the test logger inside `t.Run`
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:244]:
```suggestion
When writing a test, ensure that `logtest.Scope` in the tightest scope possible. This ensures that if a test fails, that the logging is closely tied to the test that failed. Especially if you testcase has sub tests with `t.Run`, prefer to created the test logger inside `t.Run`.
```

File: doc/dev/how-to/add_logging.md [hunk 8]
```diff
@@ -180,3 +240,11 @@ func TestFooBar(t *testing.T) {
...
+
+## Conventions
+
+* The first scope of the logger should be the name of the service and following the same naming of the service. In general, if the logger is initialized as described in [handling logging](#handling-logging) the name should be correct.
+* The logger parameter should either be immediately after `ctx context.Context`, or be the first parameter.
+* In some cases there might already be a `log` module imported. Use the alias `sglog` to refer to `lib/log` as can be seen [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/cmd/searcher/internal/search/search.go?L35-36).
```
Comment (bobheadxi) [inline doc/dev/how-to/add_logging.md:250]: Wait this is actually a bad example, both of the imports linked point to the same library! 😆

Comment (bobheadxi) [reply doc/dev/how-to/add_logging.md:250]: Maybe just provide a concrete example:

```suggestion
* In some cases there might already be a `log` module imported. Use the alias `sglog` to refer to `lib/log`, for example: `import sglog "github.com/sourcegraph/sourcegraph/lib/log"`.
```

**Review by bobheadxi — APPROVED** [review]
Nice, thanks! 😁  A few minor nits left

**Pull request merged**